app/controllers/api/v1/works_controller.rb

```diff
@@ -0,0 +1,55 @@
+class Api::V1::WorksController < Api::V1::BaseController
+  respond_to :json
+
+  # Return the URLs of a batch of individual works. Limits the number of URLs to
+  # IMPORT_MAX_CHAPTERS so it doesn't get tied up in checking URLs for too long.
+  # Params:
+  # +original_urls+:: an array of original URLs to find on the Archive
+  def batch_urls
+    original_urls = params[:original_urls]
+    status = :bad_request
+    results = []
+    if original_urls.nil? || original_urls.blank? || original_urls.size == 0
+      results << { error: "Please provide a list of URLs to find." }
+    elsif original_urls.size >= ArchiveConfig.IMPORT_MAX_CHAPTERS
+      results << { error: "Please provide no more than #{ ArchiveConfig.IMPORT_MAX_CHAPTERS } URLs to find" }
```

Comment:
My tiny useless contribution: This error message doesn't have a period at the end, but the others do.